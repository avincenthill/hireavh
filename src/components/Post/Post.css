#md-container h2,  #md-container h3, #md-container p, #md-container a, #md-container span, #md-container div{
  text-align: left;
  word-wrap: break-word;
}

#md-container h2 {
  margin-top: var(--s-m);
}

#md-container a {
  color: var(--c-4);
}

#md-container p {
  font-size: calc(var(--fs-s)*.85);
}

#md-container {
  width: max(50%, var(--s-xl));
  margin-bottom: var(--s-xl);
}