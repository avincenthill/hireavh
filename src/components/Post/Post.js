import React from "react";
// import content from "content/content";
// import Projects from "components/Projects/Projects";
import Page from "components/Page/Page";
import "./Post.css";

function Post(props) {

  return (
    <div>
      <Page>
        <h2>{props.blog.md }</h2>
      </Page>
    </div>
  );
}

export default Post;
