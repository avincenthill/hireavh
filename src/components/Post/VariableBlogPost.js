import { blogs, lookupBlogFromUrl } from 'content/blogs/blogs';
import Post from 'components/Post/Post';
import React from 'react';
import { useParams } from 'react-router-dom';

const VariableBlogPost = () => {
  const { path } = useParams();
  const blog = lookupBlogFromUrl(path, blogs);
  return blog ? <Post blog={blog}></Post> : <NotFound></NotFound>;
};

export default VariableBlogPost;
