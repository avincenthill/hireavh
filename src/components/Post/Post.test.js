import Post from "./Post";
import React from "react";
import renderer from "react-test-renderer";
import { shallow } from "enzyme";

const fakeMd = require("../../content/blogs/md/recruiter-faq.md");

describe("Post", () => {
  const fakeBlog = {
    title: "fake FAQ",
    url: "blog/fake-faq",
    description: "fake description",
    longDescription: `fake long description`,
    md: fakeMd,
    img: {
      emoji: "ðŸ¤·",
    },
  };

  describe("snapshot", () => {
    it("should match the last saved snapshot", () => {
      const tree = renderer.create(<Post blog={fakeBlog} />).toJSON();
      expect(tree).toMatchSnapshot();
    });
  });

  describe("lifecycle", () => {
    let wrapper;
    let instance;
    let fakeFetchMd;

    beforeEach(() => {
      fakeFetchMd = jest.fn(() => "<h1>fakeMd</h1>");
      wrapper = shallow(<Post blog={fakeBlog} />, {
        disableLifecycleMethods: true,
      });
      instance = wrapper.instance();
      instance.fetchMd = fakeFetchMd;
    });

    afterEach(() => {
      wrapper.unmount();
    });

    describe("componentDidMount", () => {
      it("should call fetchMd", async () => {
        await instance.componentDidMount();
        expect(fakeFetchMd).toHaveBeenCalledTimes(1);
      });
    });

    describe("render", () => {
      it("should call renderMdFile", () => {
        const renderMdString = jest.spyOn(instance, "renderMdString");
        instance.render();
        expect(renderMdString).toHaveBeenCalledTimes(1);
      });
      it("should set rendered markdown string as html", async () => {
        await instance.componentDidMount();
        // need .render() here to evaluate dangerouslySetInnerHTML
        expect(
          wrapper.find('[data-testid="md-html"]').render().text()
        ).toContain("fakeMd");
      });
    });
  });
});
