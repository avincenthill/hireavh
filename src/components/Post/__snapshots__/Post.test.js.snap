// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Post snapshot matches the last saved snapshot 1`] = `
<div>
  <div>
    <div
      className="nav nav-top"
    >
      <div
        className="nav-icon $
          this.props.isBottom ? \\"nav-icon-bottom\\" : \\"nav-icon-top\\"
        }"
      >
        <svg
          fill="currentColor"
          height="1em"
          onClick={[Function]}
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": "var(--c-1)",
              "height": "3rem",
              "width": "3rem",
            }
          }
          viewBox="0 0 8 16"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M8 4v1H0V4h8zM0 8h8V7H0v1zm0 3h8v-1H0v1z"
            fillRule="evenodd"
          />
        </svg>
      </div>
    </div>
    <a
      className="header-container"
      href="/about"
    >
      <div
        className="header-switch-theme"
      >
        <div
          className="toggle-switch"
        >
          <input
            className="toggle-switch-checkbox"
            defaultChecked={true}
            id="toggle-switch-theme"
            onChange={[Function]}
            type="checkbox"
          />
          <label
            className="toggle-switch-label"
            htmlFor="toggle-switch-theme"
          >
            <span
              className="toggle-switch-inner"
              data-no="üåô"
              data-yes="‚òÄÔ∏è"
            />
          </label>
        </div>
      </div>
      üè† about
    </a>
  </div>
  <div
    className="page-container"
  >
    <div
      id="md-container"
    />
    <hr />
    <div
      className="post-nav"
    >
      <div
        className="nav-list"
      >
        <div>
          <h2
            className="link"
          >
            <a
              className="link-a"
              href="/about"
            >
              <span>
                üè† about
              </span>
            </a>
          </h2>
        </div>
        <div>
          <h2
            className="link"
          >
            <a
              className="link-a"
              href="/blog"
            >
              <span>
                ‚úçÔ∏è blog
              </span>
            </a>
          </h2>
        </div>
        <div>
          <h2
            className="link"
          >
            <a
              className="link-a"
              href="/projects"
            >
              <span>
                üì¶ projects
              </span>
            </a>
          </h2>
        </div>
        <div>
          <h2
            className="link"
          >
            <a
              className="link-a"
              href="/resume"
            >
              <span>
                üìÑ resume
              </span>
            </a>
          </h2>
        </div>
        <div>
          <h2
            className="link"
          >
            <a
              className="link-a"
              href="/contact"
            >
              <span>
                üìá contact
              </span>
            </a>
          </h2>
        </div>
      </div>
    </div>
  </div>
  <div>
    <div
      className="nav nav-bottom"
    >
      <div
        className="nav-icon $
          this.props.isBottom ? \\"nav-icon-bottom\\" : \\"nav-icon-top\\"
        }"
      >
        <svg
          fill="currentColor"
          height="1em"
          onClick={[Function]}
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": "var(--c-1)",
              "height": "3rem",
              "width": "3rem",
            }
          }
          viewBox="0 0 8 16"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M8 4v1H0V4h8zM0 8h8V7H0v1zm0 3h8v-1H0v1z"
            fillRule="evenodd"
          />
        </svg>
      </div>
    </div>
    <a
      className="footer-container"
      href="/about"
    >
      <span>
        copyright Alex Vincent-Hill 2020 - üè† about
      </span>
    </a>
  </div>
</div>
`;

exports[`Post snapshot should match the last saved snapshot 1`] = `
<div>
  <div>
    <div
      className="nav nav-top"
    >
      <div
        className="nav-icon $
          this.props.isBottom ? \\"nav-icon-bottom\\" : \\"nav-icon-top\\"
        }"
      >
        <svg
          fill="currentColor"
          height="1em"
          onClick={[Function]}
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": "var(--c-1)",
              "height": "3rem",
              "width": "3rem",
            }
          }
          viewBox="0 0 8 16"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M8 4v1H0V4h8zM0 8h8V7H0v1zm0 3h8v-1H0v1z"
            fillRule="evenodd"
          />
        </svg>
      </div>
    </div>
    <a
      className="header-container"
      href="/about"
    >
      <div
        className="header-switch-theme"
      >
        <div
          className="toggle-switch"
        >
          <input
            className="toggle-switch-checkbox"
            defaultChecked={true}
            id="toggle-switch-theme"
            onChange={[Function]}
            type="checkbox"
          />
          <label
            className="toggle-switch-label"
            htmlFor="toggle-switch-theme"
          >
            <span
              className="toggle-switch-inner"
              data-no="üåô"
              data-yes="‚òÄÔ∏è"
            />
          </label>
        </div>
      </div>
      üè† about
    </a>
  </div>
  <div
    className="page-container"
  >
    <div
      id="md-container"
    />
    <hr />
    <div
      className="post-nav"
    >
      <div
        className="nav-list"
      >
        <div>
          <h2
            className="link"
          >
            <a
              className="link-a"
              href="/about"
            >
              <span>
                üè† about
              </span>
            </a>
          </h2>
        </div>
        <div>
          <h2
            className="link"
          >
            <a
              className="link-a"
              href="/blog"
            >
              <span>
                ‚úçÔ∏è blog
              </span>
            </a>
          </h2>
        </div>
        <div>
          <h2
            className="link"
          >
            <a
              className="link-a"
              href="/projects"
            >
              <span>
                üì¶ projects
              </span>
            </a>
          </h2>
        </div>
        <div>
          <h2
            className="link"
          >
            <a
              className="link-a"
              href="/resume"
            >
              <span>
                üìÑ resume
              </span>
            </a>
          </h2>
        </div>
        <div>
          <h2
            className="link"
          >
            <a
              className="link-a"
              href="/contact"
            >
              <span>
                üìá contact
              </span>
            </a>
          </h2>
        </div>
      </div>
    </div>
  </div>
  <div>
    <div
      className="nav nav-bottom"
    >
      <div
        className="nav-icon $
          this.props.isBottom ? \\"nav-icon-bottom\\" : \\"nav-icon-top\\"
        }"
      >
        <svg
          fill="currentColor"
          height="1em"
          onClick={[Function]}
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": "var(--c-1)",
              "height": "3rem",
              "width": "3rem",
            }
          }
          viewBox="0 0 8 16"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M8 4v1H0V4h8zM0 8h8V7H0v1zm0 3h8v-1H0v1z"
            fillRule="evenodd"
          />
        </svg>
      </div>
    </div>
    <a
      className="footer-container"
      href="/about"
    >
      <span>
        copyright Alex Vincent-Hill 2020 - üè† about
      </span>
    </a>
  </div>
</div>
`;
